orca-core/src/main/groovy/com/netflix/spinnaker/orca/config/OrcaConfiguration.groovy

```diff
@@ -145,13 +149,51 @@ class OrcaConfiguration {
   }
 
   @Bean
-  StageStatusPropagationListener stageStatusPropagationListener(ExecutionRepository executionRepository) {
-    new StageStatusPropagationListener(executionRepository)
-  }
-
-  @Bean
-  StageTaskPropagationListener stageTaskPropagationListener(ExecutionRepository executionRepository) {
-    new StageTaskPropagationListener(executionRepository)
+  CompositeStepExecutionListener compositeStepExecutionListener(ExecutionRepository executionRepository) {
+    /*
+     The CompositeStepExecutionListener runs in this order:
+     (1).beforeTask
+     (2).beforeTask
+     (2).afterTask
+     (1).afterTask
+
+     We really expect:
+     (1).beforeTask
+     (2).beforeTask
+     (1).afterTask
+     (2).afterTask
+
+     To achieve this we can overload listeners while providing no-op implementations to
+     ensure a consistent execution ordering.
+     */
+    def compositeStepExecutionListener = new CompositeStepExecutionListener()
+    compositeStepExecutionListener.setListeners([
```

Comment:
On a whim I was able to reproduce the behavior consistently by 

```
def compositeStepExecutionListener = new CompositeStepExecutionListener()
    compositeStepExecutionListener.setListeners([
      new StageTaskPropagationListener(executionRepository),
      new StageStatusPropagationListener(executionRepository)
    ] as StepExecutionListener[])
```

which led me to believe that it has something to do with the `StageStatusPropagationListener.afterTask()` (occasionally?) running before `StageTaskPropagationListener.afterTask()`.

This is not an identification of the root cause but rather an attempt to make consistent to order in which step (_aka task_) listeners execute.